src/components/TransactionStatus.tsx

```diff
@@ -0,0 +1,166 @@
+    };
+
+    const getLatestReceivedNonce = async () => {
+      const latestReceivedNonceCall = await targetApi.rpc.state.call<Codec>(latestReceivedNonceMethodName, laneId);
+      // @ts-ignore
+      const latestReceivedNonceCallType = targetApi.registry.createType('MessageNonce', latestReceivedNonceCall);
+      setLatestReceivedNonceRuntimeApi(parseInt(latestReceivedNonceCallType.toString()));
+    };
+
+    getNonceByHash();
+    getLatestReceivedNonce();
+  }, [
+    areApiLoading,
+    bestBlockFinalizedOnTarget,
+    transaction,
+    laneId,
+    latestReceivedNonceMethodName,
+    targetApi.registry,
+    targetApi.rpc.chain,
+    targetApi.rpc.state
+  ]);
+
+  useEffect(() => {
+    if (!areApiLoading || !transaction || completed) {
+      return;
+    }
+
+    const stepEvaluator = (transactionValue: string | number | null, chainValue: string | number | null) => {
+      if (!transactionValue || !chainValue) {
+        return false;
+      }
+      const bnChainValue = new BN(chainValue);
+      const bnTransactionValue = new BN(transactionValue);
+      return bnChainValue.gt(bnTransactionValue);
+    };
+
+    const completionStatus = (status: boolean) => (transaction.block && status ? 'DONE' : 'RUNNING');
+    const sourceTransactionIncluded = transaction.block
+      ? `included in block ${transaction.block}`
+      : completionStatus(false);
+    const sourceTransactionFinalized = stepEvaluator(transaction.block, bestBlockFinalized);
+    const blockFinalityRelayed = stepEvaluator(transaction.block, bestBridgedFinalizedBlockOnTarget);
+    const messageDelivered = stepEvaluator(transaction.messageNonce, latestReceivedNonceRuntimeApi);
+    const messageFinalizedOnTarget = stepEvaluator(transaction.messageNonce, nonceOfTargetFinalizedBlock);
+    const sourceConfirmationReceived = stepEvaluator(transaction.messageNonce, latestReceivedNonceOnSource);
+
+    const steps = [
+      { chainType: sourceChain, label: 'Including message in block', status: sourceTransactionIncluded },
+      {
+        chainType: sourceChain,
+        label: 'Waiting for block finality on source',
+        status: completionStatus(sourceTransactionFinalized)
+      },
+      {
+        chainType: targetChain,
+        label: 'Source Chain block finality received on target',
+        status: completionStatus(blockFinalityRelayed)
+      },
+      { chainType: targetChain, label: 'Message delivered', status: completionStatus(messageDelivered) },
+      { chainType: targetChain, label: 'Message finality', status: completionStatus(messageFinalizedOnTarget) },
+      {
+        chainType: sourceChain,
+        label: 'Delivery confirmation on source',
+        status: completionStatus(sourceConfirmationReceived)
+      }
+    ];
+
+    if (sourceConfirmationReceived) {
+      dispatchTransaction(
+        TransactionActionCreators.updateTransactionStatus({ status: TransactionStatusEnum.COMPLETED })
+      );
+    }
```

Comment:
I still don't like this as mentioned earlier. I think this is incorrect split between global (hook) and local state.

1. Transaction status is global (hook-local).
2. Tracking confirmations is local.

If I rely on the transaction status I HAVE TO include the transaction status component as well, cause otherwise I'll not receive `COMPLETED` status.

I see two possible solutions:
1. Make transactions status local (i.e. part of this component) and remove all actions and reducer for `currentTransaction`.
2. Make it possible to track MULTIPLE (ALL) transaction status globally, without including any components. I.e. this means that `COMPLETED` should be triggered by the hook and the state should not have `currentTransaction` but rather a map of `transactions: { [nonce]: status }` or sth.